test/http-api/dag.js

```diff
@@ -0,0 +1,111 @@
+/* eslint max-nested-callbacks: ["error", 8] */
+/* eslint-env mocha */
+'use strict'
+
+const chai = require('chai')
+const dirtyChai = require('dirty-chai')
+const path = require('path')
+const expect = chai.expect
+chai.use(dirtyChai)
+const DaemonFactory = require('ipfsd-ctl')
+const df = DaemonFactory.create({
+  exec: path.resolve(__dirname, '..', '..', 'src', 'cli', 'bin.js')
+})
+
+describe('dag endpoint', () => {
+  let ipfs = null
+  let ipfsd = null
+
+  before(function (done) {
+    this.timeout(20 * 1000)
+
+    df.spawn({
+      initOptions: {
+        bits: 1024
+      },
+      config: {
+        Bootstrap: [],
+        Discovery: {
+          MDNS: {
+            Enabled: false
+          },
+          webRTCStar: {
+            Enabled: false
+          }
+        }
+      }
+    }, (err, _ipfsd) => {
+      if (err) {
+        console.error(err)
+      }
+
+      expect(err).to.not.exist()
+      ipfsd = _ipfsd
+      ipfs = ipfsd.api
+      done()
+    })
+  })
+
+  after((done) => ipfsd.stop(done))
+
+  it('returns error for request without argument', (done) => {
+    ipfs.dag.get(null, (err, result) => {
+      expect(err.message).to.include("invalid 'ipfs ref' path")
+      done()
+    })
+  })
+
+  it('returns error for request with invalid argument', (done) => {
+    ipfs.dag.get('invalid', { enc: 'base58' }, (err, result) => {
+      expect(err.message).to.include("invalid 'ipfs ref' path")
+      done()
+    })
+  })
+
+  it('returns value', (done) => {
```

Comment:
Is this covered by https://github.com/ipfs/interface-js-ipfs-core/blob/8d7467b207425536b16e196ba7b12e8ce306d7fa/src/dag/get.js#L101-L114?